src/canvas-extractor/canvas_extractor/__main__.py

```diff
@@ -127,14 +130,27 @@ def extract_assignments(
 
 
 def extract_submissions(
-    assignments: List[Assignment], sync_db: sqlalchemy.engine.base.Engine
-) -> List[Submission]:
+    assignments: List[Assignment],
+    sections: List[Section],
+    sync_db: sqlalchemy.engine.base.Engine,
+):
     logger.info("Extracting Submissions from Canvas API")
-    submissions: List[Submission] = request_submissions(assignments)
-    submissions_df: DataFrame = submissions_synced_as_df(submissions, sync_db)
-    # Temporary - just for demonstration until UDM mapping
-    submissions_df.to_csv("data/submissions.csv", index=False)
-    return submissions
+    export: Dict[Tuple[str, str], DataFrame] = {}
+    for section in sections:
+        for assignment in [
+            assignment
+            for assignment in assignments
+            if assignment.course_id == section.course_id
+        ]:
+            submissions: List[Submission] = request_submissions(assignment)
+            submissions_df: DataFrame = submissions_synced_as_df(submissions, sync_db)
+            submissions_df = map_to_udm_submissions(submissions_df)
+            export[(section.id, assignment.id)] = submissions_df
+
+    SUBMISSIONS_OUTPUT_DIRECTORY = os.path.join(
+        BASE_OUTPUT_DIRECTORY, SUBMISSION_ROOT_DIRECTORY
```

Comment:
I would have expected the base output directory to come from configuration, rather than being a hard-coded constant.

Would it be possible to more closely follow the pattern of either Google Classroom or Schoology in determining the file system? In fact, the LmsFilesystemProvider from Schoology could be extracted into a shared module so that all extractors can use the same code.

Reply:
it is currently using GC approach, I have just added the .env variable support... The reason why we choose the GC over Schoology is because of GC is more generic and easier to maintain since it does not add a new method each time that we need to introduce a new path, but we can discuss about it.
@bradbanister, what do you think about this point?

Reply:
I think what's missing is the parse-args stuff in the other extractors.  I think it's in a separate story?

Reply:
That's part of FIZZ-56... the difference between the extractors regarding the filesystem, is that GC has a generic method and Schoology has a dedicated method for each file

Reply:
Missing parse-arg stuff makes sense now. I'm not in a good spot right now for judging between the two (GC/Schoology) approaches, although my reaction-without-looking-at-code is that dedicated is better than generic, as it provides transparency to end-users and future programmers.